tests/unit/test_airflow.py

```diff
@@ -0,0 +1,444 @@
+    boto_mock = Mock(name='boto_session', region_name=REGION)
+    session = Mock(name='sagemaker_session', boto_session=boto_mock,
+                   boto_region_name=REGION, config=None, local_mode=False)
+    session.default_bucket = Mock(name='default_bucket', return_value=BUCKET_NAME)
+    return session
+
+
+def test_byo_training_config_required_args(sagemaker_session):
+    byo = Estimator(image_name="byo",
+                    role="{{ role }}",
+                    train_instance_count="{{ instance_count }}",
+                    train_instance_type="ml.c4.2xlarge",
+                    sagemaker_session=sagemaker_session)
+
+    byo.set_hyperparameters(epochs=32,
+                            feature_dim=1024,
+                            mini_batch_size=256)
+
+    data = {'train': "{{ training_data }}"}
+
+    training_config = get_training_config(byo, data)
+    expected_config = {
+        'AlgorithmSpecification': {
+            'TrainingImage': 'byo',
+            'TrainingInputMode': 'File'
+        },
+        'OutputDataConfig': {
+            'S3OutputPath': 's3://output/'
+        },
+        'TrainingJobName': "byo-{{ execution_date.strftime('%Y-%m-%d-%H-%M-%S') }}",
+        'StoppingCondition': {
+            'MaxRuntimeInSeconds': 86400
+        },
+        'ResourceConfig': {
+            'InstanceCount': '{{ instance_count }}',
+            'InstanceType': 'ml.c4.2xlarge',
+            'VolumeSizeInGB': 30
+        },
+        'RoleArn': '{{ role }}',
+        'InputDataConfig': [{
+            'DataSource': {
+                'S3DataSource': {
+                    'S3DataDistributionType': 'FullyReplicated',
+                    'S3DataType': 'S3Prefix',
+                    'S3Uri': '{{ training_data }}'
+                }
+            }, 'ChannelName': 'train'
+        }],
+        'HyperParameters': {
+            'epochs': '32',
+            'feature_dim': '1024',
+            'mini_batch_size': '256'}
+    }
+    assert training_config == expected_config
+
+
+def test_byo_training_config_all_args(sagemaker_session):
+    byo = Estimator(image_name="byo",
+                    role="{{ role }}",
+                    train_instance_count="{{ instance_count }}",
+                    train_instance_type="ml.c4.2xlarge",
+                    train_volume_size="{{ train_volume_size }}",
+                    train_volume_kms_key="{{ train_volume_kms_key }}",
+                    train_max_run="{{ train_max_run }}",
+                    input_mode='Pipe',
+                    output_path="{{ output_path }}",
+                    output_kms_key="{{ output_volume_kms_key }}",
+                    base_job_name="{{ base_job_name }}",
+                    tags=[{"{{ key }}":"{{ value }}"}],
+                    subnets=["{{ subnet }}"],
+                    security_group_ids=["{{ security_group_ids }}"],
+                    model_uri="{{ model_uri }}",
+                    model_channel_name="{{ model_chanel }}",
```

Comment:
would it be worth maybe turning some of these strings into constants?

Reply:
I guess I will do it in later PR when I have Tuner/Transformer here. Then for the common arguments I just put them in constants.

Now I feel like either I put all in constants or None which is not very different.